src/main/java/dev/openfeature/sdk/EventProvider.java

```diff
@@ -0,0 +1,21 @@
+package dev.openfeature.sdk;
+
+/**
+ * Interface for attaching event handlers.
+ * 
+ * @see SimpleEventProvider for a basic implementation.
+ */
+public interface EventProvider {
+
+    /**
+     * Return the EventEmitter interface for this provider.
+     * The same instance should be returned from this method at each invocation.
+     * 
+     * @return the EventEmitter instance for this EventProvider. 
+     */
+    EventEmitter getEventEmitter();
```

Comment:
_Provider authors_ can either implement this, or extend the abstract `SimpleEventProvider`. The thing I don't like about this interface, is we rely on the instance returned by `getEventEmitter` always being the same, or things won't work. `SimpleEventProvider` works this way, but there's some reduced flexibility involved with using abstracts... so I've left both options. Interested in feedback on this.